myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/event/handlers/StatusUpdateEventHandler.java

```diff
@@ -52,7 +55,8 @@ public void onEvent(StatusUpdateEvent event,
         TaskStatus status = event.getStatus();
         this.schedulerState.updateTask(status);
         TaskID taskId = status.getTaskId();
-        if (!schedulerState.hasTask(taskId)) {
+        NodeTask task = schedulerState.getTask(taskId);
+        if (task == null) {
             LOGGER.warn("Task: {} not found, status: {}", taskId.getValue(), status.getState());
```

Comment:
Most likely task will not be there, but yes - it probably won't hurt removing it